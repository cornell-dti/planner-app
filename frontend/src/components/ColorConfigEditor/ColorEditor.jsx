// @flow strict

import * as React from 'react';
import { connect } from 'react-redux';
import { GithubPicker } from 'react-color';
import { editTag as editTagAction } from '../../store/actions';
import type { EditTagAction } from '../../store/action-types';
import type { Tag } from '../../store/store-types';

type Props = {|
  tag: Tag,
  editTag: (tag: Tag) => EditTagAction,
  changeCallback: Function,
|};

function ColorEditor(props: Props) {
  const { tag, editTag, changeCallback } = props;
  const { color } = tag;
  const handleStateComplete = (c) => {
    editTag({ ...tag, color: c.hex });
    changeCallback();
  };
  return (
    <GithubPicker color={color} onChangeComplete={handleStateComplete} triangle="top-right" />
  );
}

const ConnectedColorEditor = connect(null, { editTag: editTagAction })(ColorEditor);
export default ConnectedColorEditor;
