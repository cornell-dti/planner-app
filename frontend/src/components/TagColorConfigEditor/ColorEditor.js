import React, {Component} from 'react';
import {connect} from 'react-redux';
import {editColorConfig} from '../../store/actions.js';
import {GithubPicker} from 'react-color';

const mapDispatchToProps = dispatch => {
    return {
        editColorConfig: (tag, color) => dispatch(editColorConfig(tag, color))
    };
};

class UnconnectedColorEditor extends Component {

    handleStateComplete = color => {
        const tag = this.props.tag;
        this.props.editColorConfig(tag, color.hex);
    };

    render() {
        return (
            <GithubPicker color={this.props.color} onChangeComplete={this.handleStateComplete} />
        );
    }

}

const ColorEditor = connect(null, mapDispatchToProps)(UnconnectedColorEditor);
export default ColorEditor;
