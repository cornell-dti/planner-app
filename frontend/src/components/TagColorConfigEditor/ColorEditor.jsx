// @flow

import React from 'react';
import { connect } from 'react-redux';
import { GithubPicker } from 'react-color';
import { editColorConfig as editColorConfigAction } from '../../store/actions';

const mapDispatchToProps = dispatch => ({
  editColorConfig: (tag, color) => dispatch(editColorConfigAction(tag, color)),
});

type Props = {
  tag: string,
  color: string,
  editColorConfig: (tag: string, color: string) => void
}

class ColorEditor extends React.Component<Props> {
  handleStateComplete = (color) => {
    const { tag, editColorConfig } = this.props;
    editColorConfig(tag, color.hex);
  };

  render() {
    const { color } = this.props;
    return (
      <GithubPicker color={color} onChangeComplete={this.handleStateComplete} />
    );
  }
}

export default connect(null, mapDispatchToProps)(ColorEditor);
