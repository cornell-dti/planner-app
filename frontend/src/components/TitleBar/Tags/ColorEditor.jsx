// @flow strict

import React from 'react';
import type { Node } from 'react';
import { GithubPicker } from 'react-color';
import { Icon } from 'semantic-ui-react';
import { AngleDown } from '../../assets/svgs/v.svg'
import colMap from './ListColors';
import styles from './ColorEditor.css';

type Props = {|
  +color: string,
  +onChange: (color: string) => void;
|};
type State = {| +doesShowEditor: boolean |};

const colArray: string[] = Object.keys(colMap);

export default class ColorEditor extends React.Component<Props, State> {
  state: State = { doesShowEditor: false };

  /**
   * Toggle the editor.
   */
  toggleEditor = () => this.setState(s => ({ doesShowEditor: !s.doesShowEditor }));

  /**
   * Edit the color.
   *
   * @param {{hex: string}} e the color object from color picker.
   */
  onChangeComplete = (e: { hex: string }) => {
    const { onChange } = this.props;
    onChange(e.hex);
    this.setState({ doesShowEditor: false });
  };

  render(): Node {
    const { color } = this.props;
    const { doesShowEditor } = this.state;
    return (
      <React.Fragment>
        <button type="button" className={styles.ColorEdit} onClick={this.toggleEditor}>
          {colMap[color.toLowerCase()]}
          <span className={styles.ColorDisplay} style={{ backgroundColor: color }} />
          <AngleDown className={styles.Arrow}  />   // use AngleDown
        </button>
        {doesShowEditor && (
          <div className={styles.ColorPicker}>
            <GithubPicker
              color={color}
              onChangeComplete={this.onChangeComplete}
              triangle="top-right"
              colors={colArray}
            />
          </div>
        )}
      </React.Fragment>
    );
  }
}
