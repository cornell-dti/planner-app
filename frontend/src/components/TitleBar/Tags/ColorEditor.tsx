import React, { ReactElement } from 'react';
import { GithubPicker } from 'react-color';
import Down from '../../../assets/svgs/v.svg';
import colMap from './ListColors';
import styles from './ColorEditor.css';

type Props = {
  readonly color: string,
  readonly onChange: (color: string) => void;
};

const colArray: string[] = Object.keys(colMap);

export default function ColorEditor({ color, onChange }: Props): ReactElement {
  const [doesShowEditor, setDoesShowEditor] = React.useState(false);

  const toggleEditor = () => setDoesShowEditor(s => !s);
  const onChangeComplete = (e: { hex: string }) => {
    onChange(e.hex);
    setDoesShowEditor(false);
  };

  return (
    <React.Fragment>
      <button type="button" className={styles.ColorEdit} onClick={toggleEditor}>
        {colMap[color.toLowerCase()]}
        <span className={styles.ColorDisplay} style={{ backgroundColor: color }} />
        <Down className={styles.Arrow} />
      </button>
      {doesShowEditor && (
        <div className={styles.ColorPicker}>
          <GithubPicker
            color={color}
            onChangeComplete={onChangeComplete}
            triangle="top-right"
            colors={colArray}
          />
        </div>
      )}
    </React.Fragment>
  );
}
