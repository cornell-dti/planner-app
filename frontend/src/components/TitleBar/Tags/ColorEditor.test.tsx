import React from 'react';
import renderer from 'react-test-renderer';
import ColorEditor from './ColorEditor';

it('ColorEditor matches snapshot.', () => {
  const onChange = (): void => { };
  const tree = renderer.create(<ColorEditor color="green" onChange={onChange} />).toJSON();
  expect(tree).toMatchSnapshot();
});
