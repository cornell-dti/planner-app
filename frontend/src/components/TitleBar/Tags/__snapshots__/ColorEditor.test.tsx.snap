// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorEditor matches snapshot. 1`] = `
<button
  className="ColorEdit"
  onClick={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="ColorDisplay"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <span
    onKeyUp={[Function]}
    role="button"
    tabIndex={0}
    title="More"
  >
    <img
      alt="More"
      className="SamwiseIconDefaultStyle Arrow"
      src="/svgs/v.svg"
    />
  </span>
</button>
`;
